yatsm/pipeline/_topology.py

```diff
@@ -111,11 +114,27 @@ def _config_to_deps(config, dsk=None, overwrite=True):
                 logger.debug('Task already computed and not overwrite - not '
                              'adding: {}'.format(task))
 
-        # If this task didn't provide any new data/record, cull it
+        # TODO: maybe we don't want to cull tasks, because they
+        #           * might have important side effects (e.g., IO)
+        #           * overwrite data (e.g., masking provides and
+        #             requires `data: ['nir']`)
         if not task_needed:
```

Comment:
The original intent of this check is to remove tasks that have already been computed, usually because the computation was loaded from "cached" results on disk. The first motivation of this was to enable data computations, like NDVI, to be stored along with the "cached" time series data (another TODO still...) and we would cull tasks providing NDVI